lib/AST/ASTContext.cpp

```diff
@@ -3131,6 +3131,32 @@ AnyFunctionType *AnyFunctionType::withExtInfo(ExtInfo info) const {
   llvm_unreachable("unhandled function type");
 }
 
+AnyFunctionType *AnyFunctionType::getUncurriedFunction() {
```

Comment:
Logic for this already exists in SIL TypeLowering.cpp -- opportunity to share code here?